packages/amplify-cli-core/src/state-manager/pathManager.ts

```diff
@@ -146,7 +147,7 @@ export class PathManager {
   getResourceParametersFilePath = (projectPath: string | undefined, category: string, resourceName: string): string => {
     let isBuildParametersjson: boolean = false;
     const resourceDirPath = this.getResourceDirectoryPath(projectPath, category, resourceName);
-    if (!fs.existsSync(path.join(resourceDirPath, PathConstants.ParametersJsonFileName))) {
+    if (!fs.existsSync(path.join(resourceDirPath, PathConstants.ParametersJsonFileName)) && overriddenCategories.includes(category)) {
```

Comment:
API categories also have parameters.json in build

Reply:
It directly read and writes to resource